packages/timeline-state-resolver/src/integrations/vmix/vMixResponseStreamReader.ts

```diff
@@ -0,0 +1,104 @@
+import { EventEmitter } from 'eventemitter3'
+
+type ResponseStreamReaderEvents = {
+	response: [response: Response]
+	error: [error: Error]
+}
+
+export interface Response {
+	command: string
+	response: 'OK' | 'ER'
+	message: string
+	body?: string
+}
+
+const RESPONSE_REGEX = /^(?<command>\w+)\s+(?<response>OK|ER|\d+)(\s+(?<responseMsg>.*))?/i
+
+/**
+ * A receiver for vMix responses
+ */
+export class VMixResponseStreamReader extends EventEmitter<ResponseStreamReaderEvents> {
+	private _unprocessedLines: string[] = []
+	private _lineRemainder = ''
+
+	reset() {
+		this._unprocessedLines = []
+		this._lineRemainder = ''
+	}
+
+	processIncomingData(data: Buffer) {
+		const string = this._lineRemainder + data.toString('utf-8')
+		this._lineRemainder = ''
+		const lines = string.split('\r\n')
+
+		if (lines[lines.length - 1] === '') {
```

Comment:
I feel like this could be written simpler by doing a `const lastNewlineIndex = string.lastIndexOf('\r\n')`.
then it would be something like:
```
const lastNewlineIndex = string.lastIndexOf('\r\n')
if (lastNewLineIndex === -1) {
  this._lineRemainder = string
  return
} else {
  const lines = string.slice(0, lastNewLineIndex).split('\r\n')
  this._unprocessedLines.push(...lines)
  this._lineRemainder = string.slice(lastNewLineIndex + 2)
}

```
To me this feels more natural, but maybe it isnt any simpler actually.

Reply:
I like it. nice and short and easy to follow